File: src/main/java/com/ts/advices/infrastructure/dao/AuthorRepository.java
package com.ts.advices.infrastructure.dao;

import com.ts.advices.infrastructure.vo.AuthorVO;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface AuthorRepository extends CrudRepository<AuthorVO, Long> {

    AuthorVO findByUserName(String userName);
}

File: src/main/java/com/ts/advices/infrastructure/dao/AuthorRepository.kt
package com.ts.advices.infrastructure.dao

import com.ts.advices.infrastructure.vo.AuthorVO
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface AuthorRepository : CrudRepository<AuthorVO, Long> {
    fun findByUserName(userName: String): AuthorVO
}
